src/Symfony/Component/Translation/TranslatorInterface.php

```diff
@@ -36,7 +36,7 @@ public function trans($id, array $parameters = array(), $domain = null, $locale
      * Translates the given choice message by choosing a translation according to a number.
      *
      * @param string      $id         The message id (may also be an object that can be cast to string)
-     * @param int         $number     The number to use to find the indice of the message
+     * @param float|int   $number     The number to use to find the indice of the message
```

Comment:
This is a BC break, as it changes the interface (what about implementations not accepted floats today ? they would not respect the interface anymore).

And this would be a bad message IMO, as floats are not fully supported in your PR.